modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Parts/Search/SearchStrategies/BaseSearch.php

```diff
@@ -428,13 +456,60 @@ public function runSimpleParamsSearch()
     public function runMysqlQuery()
     {
         $dataFromDb = $this->searchThisTable->buildGenericSelect($this->searchParams, implode(",", $this->orderParams), array());
+        $includeFlag= !empty($this->includeParams);
         foreach ($dataFromDb as $key => $data) {
+            if($includeFlag){
+                foreach ($this->includeParams as $index => $include){
+                    if(!empty($include[1]) && !empty($data[$include[1]]) ){
+                        $this->includeParams[$index][3]=$data[$include[1]];
+                    }
+                }
+            }
             $this->fhirObj->initFhirObject();
             $FHIRResourceContainer = new FHIRResourceContainer($this->fhirObj->DBToFhir($data));
             $this->FHIRBundle = $this->fhirObj->addResourceToBundle($this->FHIRBundle, $FHIRResourceContainer, 'match');
-            // $this->FHIRBundle->deletePatient();
         }
+
+        if($includeFlag){
+            foreach ($this->includeParams as $index => $include){
+                $FHIRElm=$include[2];
+                $readId= $include[3];
+                if(!empty($FHIRElm) && !empty($readId)){
+                    $class = self::FHIR_ELM_PATH.$FHIRElm."\\".$FHIRElm;
+                    if(class_exists($class)){
+                        $initials=array();
+                        $initials['paramsFromUrl']=array("0"=>1);
+                        $initials['paramsFromBody']=array();
+                        $initials['container']=$this->container;
+                        $FHIRElmClass=new $class($initials);
+                        $FHIRElmOBJ=$FHIRElmClass->read();
```

Comment:
we can write it more effectively as we talked. instead of read every entity from the DB. is it not better?
1. collect all the id's you want to include into array
2. search all the entities from the DB using  $this->searchOfIncludeTable->buildGenericSelect(array('id' => 2,'id' => 7, 'id' =>8))
3. loop with DBToFhir and addResourceToBundle

( I just don't know how we know what is the "id" column of every entity)